### PR TITLE
fix(i18n): update Chinese translation of basic html and html5

### PR DESCRIPTION
**Pull request by ZhichengChen**
- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [X] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: @raisedadead Should we translate the example snippet into Chinese as well (More specifically, "this button submits the form" here)? Personally, I'd consider that makes more sense by having it translated.

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: nit: incorrect indent

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: orange cat: 橘猫

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: Lasagna: [千层面](https://zh.wikipedia.org/wiki/%E5%8D%83%E5%B1%A4%E9%BA%B5)。这个梗的来源是[加菲猫](https://zh.wikipedia.org/wiki/%E5%8A%A0%E8%8F%B2%E7%8C%AB)

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: flea 是“跳蚤”，但 flea treatment 是“除跳蚤”或者“祛跳蚤”。你选一个吧😂

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-a-submit-button-to-a-form.chinese.md]: 我觉得这里用 `URL` 会比 "地址" 更好。你觉得呢？

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md]: 除非需要 => 除非必要

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md]: `assert` 的第二个参数，印象中是 failure 的时候显示的内容。应该也需要翻译吧？和上面一样就好

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md]: 这个描述我觉得可能不够具体，因为 test 内容是 `img` 的 `src`。你觉得有必要描述得详细一些么？

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md]: 应该是“囤老鼠”吧？
另外，"猫咪慎入，"感觉可以去掉

**Review by S1ngS1ng — CHANGES_REQUESTED** [review]
@raisedadead Here are some questions to you:
1. Should we translate the text in the code sample as well? (IMO, it is better to have them translated).
2. I noticed that all the contents of `Solution` have been removed. Is that as planned?
3. Looks like the single quotes of `Tests/tests/textString` have been removed, while a pair of single quotes have been added to `Tests/tests/text`. Is it fine?

@ZhichengChen Just reviewed the first file. Will review the rest tomorrow.

**Comment by raisedadead** [comment]
> @raisedadead Here are some questions to you:
> 
> 1. Should we translate the text in the code sample as well? (IMO, it is better to have them translated).
> 2. I noticed that all the contents of `Solution` have been removed. Is that as planned?
> 3. Looks like the single quotes of `Tests/tests/textString` have been removed, while a pair of single quotes have been added to `Tests/tests/text`. Is it fine?
> 

Hi @RandellDawson can you guide us here when you get a minute.

**Comment by RandellDawson** [comment]
> Should we translate the text in the code sample as well? (IMO, it is better to have them translated).

If you do it there, will you be translating text in the actually challenge seed code?  If so, you may also have to update the tests that may be testing for a specific string.  @raisedadead I personally think the code text should remain in English and then we figure out a way to leverage the solution contained in the English version to use as the solution for each language.  This way, the only thing the `textString` would not have to change at all and only the `text` would be translated where applicable.

> I noticed that all the contents of Solution have been removed. Is that as planned?

If there are solutions missing, I am not sure that was intentional.  However, my response above is still my recommendation for the solutions across all languages.

> Looks like the single quotes of Tests/tests/textString have been removed, while a pair of single quotes have been added to Tests/tests/text. Is it fine?

If you take a look at the English versions of the same challenge (which you should do anything to validate the translation is correct and there are not missing or changed `text` or `testString`), you will see how to properly format each.  

You will also note that the English version for all but the back-end challenges and React challenges does not include text for an assert message.  That is because the `text` is used for the assert message.  As long as you are using the English version as a guide for each challenge, you can see when to include them and when to not include them.

@ojeytonwilliams Any thoughts on how we can parse the English solution(s) out to use for each language?  This would simplify the non-English versions and keep them on track with the English version of the challenge.z

@raisedadead If we decide to let each language stand alone (which has also been discussed in the past), then the solutions would be unique to each language and then it would make sense to allow the translation of the text in code and the challenge seed and solutions.  It is really your call.  I have already stated my preference.

**Comment by raisedadead** [comment]
I think the seed code should be in english. We should be able to use the script that you have to populate the Chinese curriculum after these PRs have landed.

**Comment by ojeytonwilliams** [comment]
> Any thoughts on how we can parse the English solution(s) out to use for each language? This would simplify the non-English versions and keep them on track with the English version of the challenge.

What's the objective?  Extract the English solutions and overwrite the solutions for the other languages?  It sounds like you already have a script for that, so I assume I'm missing something.

**Comment by RandellDawson** [comment]
@raisedadead If the seed code is in English, then if we parse out the solution(s) from the English version of the same challenge, then the testStrings would not need translation nor would we need to copy over the solutions.

**Comment by RandellDawson** [comment]
> What's the objective? Extract the English solutions and overwrite the solutions for the other languages? It sounds like you already have a script for that, so I assume I'm missing something.

No, I mean the non-English files would not have a solutions section or it could be empty because the challenge parser would get them from the English version.

**Comment by raisedadead** [comment]
You are correct, the parser logic would need some work.

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md
Comment (ZhichengChen) [reply curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/add-images-to-your-website.chinese.md]: 在英文的最新的版本里去掉了第二个参数，这里是个冗余参数，我在本地测试过，没有第二个参数也是 OK 的

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/uncomment-html.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/uncomment-html.chinese.md]: 上午讨论的，如果之前那道题目要改，这里也得同步

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-html5-to-require-a-field.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-html5-to-require-a-field.chinese.md]: “加上required属性”
required 两边应该有空格

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-html5-to-require-a-field.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-html5-to-require-a-field.chinese.md]: 给\<code>input\</code>元素加上\<code>required\</code>属性，这样用户就必须先在输入框里填入内容，然后才可以提交表单。

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: 当表单提交时，包括 options 已选值在内的数据会发送给服务端。<code>radio</code>和<code>checkbox</code>的<code>value</code>值决定了发送到服务端的实际内容。

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: \<code> 标签周围应该没有空格吧？下同

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: 给每一个\<code>radio\</code>和\<code>checkbox\</code>输入框添加\<code>value\</code>属性。请把每个\<code>input\</code>对应的\<code>label\</code>文本转换为小写（如 Outdoor 应转换为 outdoor），设置其为 value 的值（即 \<code>value="outdoor"\</code>）。

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md
Comment (S1ngS1ng) [inline curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: 这部分是不是应该去掉？

**Review by S1ngS1ng — CHANGES_REQUESTED** [review]
@ZhichengChen 
1. 建议保留 \<code>a\</code>，而不翻译为“锚点”。see: https://developer.mozilla.org/zh-CN/docs/Web/HTML/Element/a 。如果要改，记得全局查找替换下，测试文本里也有
2. 有几个地方的空格需要调整下

其他没啥问题

@raisedadead 90% of **THIS** PR is good to go. Still, there are some nitpicks to be fixed. Should be ready in a day or two.
I synced with @ZhichengChen this morning. We've decided to collaborate on the translation both "online" and "offline". 
The "online" refers to the ~~conventional~~ or ~~canonical~~ way that we bot post our review comments here at GitHub. Hopefully, you guys are fine when we post the review comments in Chinese, since this makes our review process way easier. We'll make sure to post the comments in English when we consider it related to other members/contributors.
The "offline" works in a way that we both hop over on a session (e.g. Hangout, Zoom.us) to review the PR, which should be ready to ship right after. Due to the timezone, this may happen once/twice a week. 

Since the review process could take a bit while, hopefully, it won't block the release of the next version.

In case of any questions/comments/concerns, please let us know. Thanks!

File: curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md
Comment (ZhichengChen) [reply curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: 这个地方我昨天忘记和你说了，新版 css 做了调整，<code> 标签周围是有空格的，
https://raw.githubusercontent.com/freeCodeCamp/freeCodeCamp/master/curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.english.md
个人觉得可以添加空格，这样不管样式如何改变，显示效果不会有太大的影响。
另外好多 testString 上面也有 <code> 标签，如果不加空格的话显示会有点问题。
另外 <code> 标签页大部分没有添加空格，数目比较大，我觉得可以优先级调低。后期在慢慢统一。

Comment (S1ngS1ng) [reply curriculum/challenges/chinese/01-responsive-web-design/basic-html-and-html5/use-the-value-attribute-with-radio-buttons-and-checkboxes.chinese.md]: @ZhichengChen 好的，没问题

**Review by S1ngS1ng — APPROVED** [review]
LGTM 👍

**Review by S1ngS1ng — APPROVED** [review]
LGTM 👍

**Comment by ZhichengChen** [comment]
@S1ngS1ng need approve to merge

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
